worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -629,10 +631,18 @@ public void sort() throws InterruptedException {
           for (ShuffleBlockInfo blockInfo : originShuffleBlocks) {
             long offset = blockInfo.offset;
             long length = blockInfo.length;
-            ShuffleBlockInfo sortedBlock = new ShuffleBlockInfo();
-            sortedBlock.offset = fileIndex;
-            sortedBlock.length = length;
-            sortedShuffleBlocks.add(sortedBlock);
+            // combine multiple small length `ShuffleBlockInfo` for same mapId such that
+            // size of compacted `ShuffleBlockInfo` does not exceed `shuffleChunkSize`
+            if (!sortedShuffleBlocks.isEmpty()
+                && sortedShuffleBlocks.get(sortedShuffleBlocks.size() - 1).length + length
```

Comment:
can we introduce a variable for `sortedShuffleBlocks.get(sortedShuffleBlocks.size() - 1)`

Reply:
refactored the code a little to simplify it.